nixos/modules/services/misc/ersatztv.nix

```diff
@@ -0,0 +1,126 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+
+let
+  inherit (lib)
+    mkIf
+    getExe
+    maintainers
+    mkEnableOption
+    mkOption
+    mkPackageOption
+    ;
+  inherit (lib.types) str path bool;
+  cfg = config.services.ersatztv;
+in
+{
+  options = {
+    services.ersatztv = {
+      enable = mkEnableOption "ErsatzTV";
+
+      package = mkPackageOption pkgs "ersatztv" { };
+
+      user = mkOption {
+        type = str;
+        default = "ersatztv";
+        description = "User account under which ErsatzTV runs.";
+      };
+
+      group = mkOption {
+        type = str;
+        default = "ersatztv";
+        description = "Group under which ErsatzTV runs.";
+      };
+
+      transcodeDir = mkOption {
+        type = path;
+        default = "/var/lib/etv-transcode";
+        description = ''
+          Temporary transcode directory
+        '';
+      };
+
+      configDir = mkOption {
+        type = path;
+        default = "/var/lib/ersatztv";
+        description = ''
+          Directory containing the server configuration files and SQLite Database file
+        '';
+      };
```

Comment:
Do we _need_ it to be configurable? I'd just make it be (a subfolder of) the state directory.

Reply:
Generally having any directory configurable is _very_ nice when running it under any kind of impermanence setup :)

Reply:
There doesn't seem to be any standard for it in nixpkgs, and many many modules supports this https://search.nixos.org/options?channel=unstable&query=statedir . What's the downside of it?

Reply:
it is mainly broken up like that because that is [how their Dockerfile splits it](https://github.com/ErsatzTV/ErsatzTV/blob/272f528f7a21c5b18f784cd9c77c262ab3d694ad/docker/Dockerfile#L46-L47)

Reply:
So what exactly are you recommending I change @ambroisie? A single configuration option and just specify the two needed environment variables based on that? I'm not sure it makes sense to add a forced structure like that when the package itself doesn't have that requirement.

Reply:
I asked in contributors, and it seems like what @ambroisie is saying would be pretty much the way to go.

Context: https://matrix.to/#/!CTCrFzsBPYmDLmrja4:0upti.me/$nx1t-17R30Fs2nDCTJeZ0SN2o273qlj5He_oXx5DAbQ?via=nixos.org&via=matrix.org&via=tchncs.de

Reply:
Can we get resolved this as the last thing? It would be cool to get merged. @allout58 if you're all out of spoons I might be able to pick this up and get it the last way :)

Reply:
I think the change I've made is what @ambroisie had suggested? I wasn't 100% clear on the requested change, so I might be wrong.